BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2776,6 +2779,54 @@ public override Vector3 GetForwardTransform()
                 return MissileReferenceTransform.forward;
         }
 
+        public override float GetKinematicTime()
+        {
+            // Get time at which the missile is traveling at the GetKinematicSpeed() speed
+            if (!launched) return -1f;
+
+            float missileKinematicTime = boostTime + cruiseTime + cruiseDelay + dropTime - TimeIndex;
+            float speed = currentThrust > 0 ? optimumAirspeed : (float)vessel.srfSpeed;
+            float minSpeed = GetKinematicSpeed();
+            if (speed > minSpeed)
+            {
+                float airDensity = (float)vessel.atmDensity;
+                float dragTerm;
+                float t;
+                if (useSimpleDrag)
+                {
+                    dragTerm = (deployed ? deployedDrag : simpleDrag) * (0.008f * part.mass) * 0.5f * airDensity;
+                    t = part.mass / (minSpeed * dragTerm) - part.mass / (speed * dragTerm);
+                }
+                else
+                {
+                    float AoA = smoothedAoA.Value;
+                    FloatCurve dragCurve = MissileGuidance.DefaultDragCurve;
+                    float dragCd = dragCurve.Evaluate(AoA);
+                    float dragMultiplier = BDArmorySettings.GLOBAL_DRAG_MULTIPLIER;
+                    dragTerm = 0.5f * airDensity * dragArea * dragMultiplier * dragCd;
+                    float dragTermMinSpeed = 0.5f * airDensity * dragArea * dragMultiplier * dragCurve.Evaluate(Mathf.Min(29f, maxAoA)); // Max AoA or 29 deg (at kink in drag curve)
+                    t = part.mass / (minSpeed * dragTermMinSpeed) - part.mass / (speed * dragTerm);
+                }
+                missileKinematicTime += t; // Add time for missile to slow down to min speed
+            }
+
+            return missileKinematicTime;
+        }
+
+        public override float GetKinematicSpeed()
+        {
```

Comment:
This should probably just return 0 if the vessel is in a vacuum / near vacuum, which would avoid the need for those checks where this is being called and avoid `kinematicSpeed` becoming `Inf`.

Also, if `liftArea` is 0 (e.g., for bombs), then this is going to return half the speed of sound. Is that a problem?